gossip/server.go

```diff
@@ -88,6 +88,10 @@ func (s *server) Gossip(argsI proto.Message) (proto.Message, error) {
 	if err != nil {
 		return nil, util.Errorf("local addr %s could not be converted to net.Addr: %s", args.LAddr, err)
 	}
+	// Verify that the client connection is valid and hasn't been closed.
+	if _, ok := s.lAddrMap[lAddr.String()]; !ok {
```

Comment:
The trouble is the close callback: there is no way to tie the close callback (where all we have is the network-level connection details) to the Gossip RPC handler (where we have the actual args). We'd need an alternate version of the RPC interface (kind of like `RegisterAsync`) that passes the connection object (or an opaque connection ID) into the RPC handler.